pkgs/applications/virtualization/gvisor/default.nix

```diff
@@ -0,0 +1,176 @@
+{
+  stdenv
+, pkgs
+, fetchFromGitHub
+, cacert
+, symlinks
+, writeScript
+
+, coreutils
+, bash
+, bazel
+, git
+, go
+, python
+}:
+
+let
+
+  # gvisor doesn't have any releases yet
+  version = "2018-11-10";
+
+  # From the Bazel docs:
+  #   Beneath the outputUserRoot directory, we also create an outputBase
+  #   directory whose name is the MD5 hash of the path name of the workspace
+  #   directory.
+  #
+  # The default `fetchzip` arguments will unpack our file as the name `source`,
+  # and the Nix sandbox builds things in the `/build` directory. Thus, we can
+  # predict this hash and calculate it here:
+  outputBaseHash = builtins.hashString "md5" "/build/source";
+
+  # Source from GitHub with additional patches.
+  patchedSource = fetchFromGitHub {
+    owner = "google";
+    repo  = "gvisor";
+    rev   = "d97ccfa346d23d99dcbe634a10fa5d81b089100d";
+
+    # NOTE: this is the output of the whole fixed-output derivation, so
+    # `nix-prefetch-git` won't work to obtain this. The easiest way is to just
+    # change it and see what breaks :)
+    sha256 = "1bcnq7kazbf6l5j0g82x2lvg1nbp7z70klk139dxi0jkw0j8dh3r";
+
+    # Patch the source to:
+    #   - Use our host Go toolchain
+    #   - Fix hard-coded paths to Bash, coreutils, etc.
+    #   - Fetch a specific version of abseil instead of the `master` branch
+    extraPostFetch =
+      let
+        # master as of 2018-11-10
+        abseilRev  = "070f6e47b33a2909d039e620c873204f78809492";
+        abseilHash = "a4298dca4149157b379588bec19493bcab56f8b3fb119ea81303b04d70af1b48";
+
+      in ''
+        substituteInPlace "$out/WORKSPACE" \
+          --replace 'go_register_toolchains(go_version="1.11.2")' 'go_register_toolchains(go_version="host")' \
+          --replace \
+            'urls = ["https://github.com/abseil/abseil-cpp/archive/master.zip"],' \
+            'urls = ["https://github.com/abseil/abseil-cpp/archive/${abseilRev}.zip"], sha256 = "${abseilHash}",'
+
+        find "$out" -name '*.sh' -exec \
+          sed -i 's|#!/bin/bash|#!${bash}/bin/bash|g' {} \;
+
+        find "$out" -name '*.go' -exec \
```

Comment:
It seems this is only required by tests. Could you try with `'*_test.go'`?
Also, it woul be nice to patch this upstream:/

Reply:
I'll look into patching this upstream at some point, sure. For now, fixed this, the merge conflict, and force-pushed.